### ISSUE TITLE
CSS bundling no longer working as it used to be with Rollup 2

### ISSUE THREAD

**Issue by habibrosyad** [sveltejs/sapper-template#233]
Hi, I'm using https://github.com/sveltejs/sapper-template-rollup as the base of my project months ago. Today, I updated packages as well as the rollup configs according to the latest commit of the template (particularly updating rollup from v1 to v2). However, I noticed that when building the app, all my css bundled into a single file only (main.css) whereas previously there were several css files (I guess it's based on the routes). Currently this behaviour breaks some of the styling in my project because there are conflicts between CSS rules.

Is there away to mitigate this issue (apart from rolling back prior to the update)?

An example for the list of css files generated with the vanilla dev build of the template (prior to update):
```
[slug].8afac42c.css
client.6bfaa544.css
index.7be4ff0d.css
index.9bb43f04.css
main.237455804.css
```
After update:
```
main.2065939480.css
```

The main reason that I noticed seems to be because of upgrading rollup v1 to v2. May need to adjust the rollup config, but I have no idea what to change.

**Comment by benmccann** [comment]
I can confirm this. When visiting http://localhost:3000/profile/@ikzsl5 in the realworld sample it included http://localhost:3000/client/chunk.aef098f1.css until https://github.com/sveltejs/realworld/pull/55

I have no idea whether this change was intended or how to affect the behavior. You'll probably have better luck asking questions about on Rollup on https://gitter.im/rollup/rollup or https://stackoverflow.com/questions/tagged/rollupjs or filing issues in https://github.com/rollup/rollup/issues. Please keep us updated in this issue though with any feedback or suggestions you receive from the Rollup community

**Comment by habibrosyad** [comment]
Ok, will do.

**Comment by habibrosyad** [comment]
@benmccann per the response in rollup/rollup#3655, they suggest that Sapper might not have been updated properly to support Rollup 2, therefore the breaking changes. Should this issue be brought up to Sapper instead?

**Comment by benmccann** [comment]
Hmm. So I investigated more and I'm not quite sure why visiting http://localhost:3000/profile/@ikzsl5 in the realworld sample would include http://localhost:3000/client/chunk.aef098f1.css with Rollup 1 because the only `style` tag is in `_error.svelte`, so it actually seems maybe the Rollup 1 behavior was broken in some way too.

I think perhaps it will be better to investigate using the `hn.svelte.dev` sample

**Comment by habibrosyad** [comment]
With the `hn.svelte.dev` sample and downgrading the Rollup to version 1 I could get different CSS being loaded when you visit different routes:

Visiting http://localhost:3000/top/1
```html
<link rel="stylesheet" href="client/main.1788086292.css">
<link rel="stylesheet" href="client/[page].b543e849.css">
<link rel="stylesheet" href="client/client.cdab43ac.css">
```

Visiting http://localhost:3000/item/23703367
```html
<link rel="stylesheet" href="client/main.1788086292.css">
<link rel="stylesheet" href="client/[id].465e0571.css">
<link rel="stylesheet" href="client/client.cdab43ac.css">
```

I suppose `main` and `client` seems to be the common CSS shared between those routes. Might as well unnecessary styles being loaded as you suggest there (weird behaviour with Rollup 1). However, with Rollup 2 everything just become a single file. Therefore, I think the later is much worse than the first one.

**Comment by benmccann** [comment]
In Rollup 2, `chunk.modules` no longer contains the `.css` files and so the CSS files never get written to disk. I'm not sure whether there's anything Sapper can do about this. It seems most likely to need to be addressed in either Rollup or rollup-plugin-svelte

https://github.com/sveltejs/sapper/blob/be8db302fa58d4186b6ea0e03021e8cfe6f98348/src/core/create_compilers/extract_css.ts#L143

**Comment by benmccann** [comment]
Stab in the dark - I wonder if https://github.com/sveltejs/rollup-plugin-svelte/pull/72 has any chance of fixing it

**Comment by benmccann** [comment]
sveltejs/rollup-plugin-svelte#72 doesn't help. It still looks like a good change. But this must be caused by something else

**Comment by benmccann** [comment]
If I set `treeshake: false` then it works the way it did with Rollup 1

**Comment by habibrosyad** [comment]
As summary based on discussion in https://github.com/rollup/rollup/issues/3655, another way to fix this without disabling treeshake feature would be using https://github.com/rollup/rollup/pull/3663 (may need to wait for this to be released into master first) and update Sapper codes in https://github.com/sveltejs/sapper/blob/be8db302fa58d4186b6ea0e03021e8cfe6f98348/src/core/create_compilers/RollupCompiler.ts#L41-L46 into:
```javascript
transform: (code: string, id: string) => {
  if (/\.css$/.test(id)) {
    this.css_files.push({ id, code });
    return {code: ``, moduleSideEffects: 'no-treeshake'};
  }
}
```

**Comment by habibrosyad** [comment]
@benmccann Should I make a PR for this in Sapper? I've checked that those changes do not affect anything when running sapper project with the current template.

**Comment by benmccann** [comment]
Cool. Let's wait until https://github.com/rollup/rollup/pull/3663 is merged to confirm that will be the API, but then yeah it will make sense to make that change here

**Comment by benmccann** [comment]
I also tested and confirmed the fix. I sent a PR here: https://github.com/sveltejs/sapper/pull/1306

**Comment by habibrosyad** [comment]
Thanks @benmccann, also https://github.com/rollup/rollup/pull/3663 has been released in rollup@2.21.0, so the only things remains is for https://github.com/sveltejs/sapper/pull/1306 to get merged.

**Comment by lukeed** [comment]
Closed via https://github.com/sveltejs/sapper/pull/1306 – thank you @benmccann
